### DISCUSSION TITLE
when can we expect production release ?

### DISCUSSION THREAD

**Discussion by harishkrishnan24** [UI5/webcomponents#3307] [General]
Also, is there any plans to support Typescript ?

**Comment by pskelin** [comment]
In terms of quality, we consider the current version production-ready (hence the release candidate in the version string).

We are currently in the process of making API reviews of components to discuss the best and most consistent API names. Once this is ready, we will release a 1.0 version after which there will be no more breaking changes. Even with these planned API changes, the update effort on your side should not be big, so if you have a use case you can safely be using ui5 webcomponents right now.

As for typescript, there are such plans as well. Could you help with specific use cases?

The main issue is that typescript checking works only in typescript code, but the web component consumption happens in templates (vuejs templates, angular templates). Such templates are not TS files and there is no type checking or code completion for them. So for the overall web component story, typing is a bit less necessary.

Hence the question to provide more specific use cases and code - how do you consume web components, where do you expect types to be present?

**Reply by harishkrishnan24** [reply]
I am just expecting typings file like mentioned in https://github.com/SAP/ui5-webcomponents/issues/2470. I just found this. So guess you have answered my question. :)

**Reply by nemethmik** [reply]
Hi Peter!
Let me share my thoughts, too. When you work with Angular, you work with TypeScript.
As a solution architect for my corporation, I am in the process of evaluating SAP UI5 Web Components for new applications, where we'd love to use **Vite**, **Lit** and **TypeScript**. We are a TypeScript workshop, we are not starting plain old JavaScript projects since three years now. Actually, I have created [all the necessary d.ts files](https://github.com/nemethmik/paola18/tree/master/src/ui5types) as I needed them using the TypeScript compiler itself with a series of commands like `tsc ./node_modules/@ui5/webcomponents/src/Dialog.js --declaration --allowJs --emitDeclarationOnly --outDir ./src/ui5types` My observation was that these series of commands revealed  issues with your code: missing JSDoc details, too many `any`, and sometimes inconsistencies in the function overriding in the class hierarchy. I highly recommend that you make **d.ts** files just for quality purposes, too. You wrote that `We are currently in the process of making API reviews of components`; I'd highly recommend to use `tsc --declaration` as a tool that would help you tremendously detect issues. With this you hit two *flies with one hit*: (much) better quality code, and providing typings for TypeScript customers. 
Your product is awesome, it deserves special care.
Take care and thank you for your work, Miklos
